AddOns/rr_mapStuff_main/fncs/mapEH.sqf

```diff
@@ -30,30 +30,40 @@ if (_mapIsOpened) then {
 	private _startAnim = ["RR_gesture_transitionToMapStand","RR_gesture_holdMapProne"] select _isProne;
 	private _mainAnim  = ["RR_gesture_holdMapStand","RR_gesture_holdMapProne"] select _isProne;
 	ace_player playAction _startAnim;
-	private _map = "RR_map_handheld" createVehicle [-1,-1,-1];
+	
+	switch (worldname) do {
+		case "Altis": {
+			map = "Land_Map_Unfolded_Altis_F" createVehicle [-1,-1,-1];
+		};
+		case "Malden": {
+			map = "Land_Map_Unfolded_Malden_F" createVehicle [-1,-1,-1];
+		}; 
+		case "Tanoa": {
+			map = "Land_Map_Unfolded_Tanoa_F" createVehicle [-1,-1,-1];
+		}; 
+		case "Enoch": {
+			map = "Land_Map_Unfolded_Enoch_F" createVehicle [-1,-1,-1];
+		}; 
+		default {
+			map = "Land_Map_Unfolded_F" createVehicle [-1,-1,-1];
+		}; 
+	};
+	
+	private _map = map;
```

Comment:
```suggestion
	
	private _mapName = switch (worldname) do {
		case "Altis": {"Land_Map_Unfolded_Altis_F"};
		case "Malden": {"Land_Map_Unfolded_Malden_F"};
		case "Tanoa": {"Land_Map_Unfolded_Tanoa_F"};
		case "Enoch": {"Land_Map_Unfolded_Enoch_F"};
		default {"Land_Map_Unfolded_F"};
	};
	
	private _map = _mapName createVehicle [-1,-1,-1];
```
Global variable should be avoided. Not tested.

Reply:
Tested and works